Bit depths DSEGOG-375

joshdimanteto opened the pull request:
## Description
- The bit_depth is supplied in the channel metadata (records endpoint) this should be forwarded to the image api hooks so that the correct bit depth is used to fetch for the image 

## Testing instructions
Add a set up instructions describing how the reviewer should test the code

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage
- [ ] {more steps here}

## Agile board tracking
closes DSEGOG-375

joshdimanteto committed:
update the types

joshdimanteto committed:
Implement bit depth to colour and colourbars

joshdimanteto committed:
e2e tests

joshdimanteto committed:
create e2e test for a real 12 bit image

louise-davies commented on e2e/mocked/images.spec.ts:
```diff
@@ -145,7 +145,7 @@ test('user can change the false colour parameters of an image', async ({
   });
 
-  await expect(slider.nth(0)).toHaveValue(`${0.4 * 255}`);
+  await expect(slider.nth(0)).toHaveValue('1635');
```
Could this not continue to be calculated like `0.4 * 4095`?

joshdimanteto replied:
I did try that initially, but that value gives '1638' instead of '1635', causing the test to fail

louise-davies replied:
That's odd that it's no longer exact anymore oh well

louise-davies commented on e2e/real/images.spec.ts:
```diff
@@ -143,6 +143,137 @@ test('user can change the false colour parameters of an image', async ({
+
+  await page.getByRole('button', { name: 'Search', exact: true }).click();
+
+  // add trace channel to the table so we can click on a trace
```
```suggestion
  // add image channel to the table so we can click on an image
```

louise-davies reviewed (changes requested):
This works fine when clicking an image from the table, but when using the thumbnail selector it does not change bit depth correctly. Need to also edit `windows/ThumbnailSelector` to pass the potential new bit depth alongside the new record id.

joshdimanteto committed:
update comment

joshdimanteto committed:
supply the bit depth in the window thumbnail view

joshdimanteto commented:
> This works fine when clicking an image from the table, but when using the thumbnail selector it does not change bit depth correctly. Need to also edit `windows/ThumbnailSelector` to pass the potential new bit depth alongside the new record id.

Good spot. I've addressed this in the latest commit

joshdimanteto committed:
Merge branch 'develop' into bit_depths

* develop:
  Annotate digests with exact version - this helps keep track of what version a SHA actually is
  Log API output on CI job failure
  Fix poetry cache key in real e2e testing CI job
  Update CI workflow file based on changes to the og-api workflow file
  Remove react router v7 future flags
  Remove unneeded eslint/compat & tidy up eslint.config.js
  Update tanstack-query monorepo to v5.68.0
  Update dependency lint-staged to v15.5.0
  Update docker/login-action action to v3.4.0
  Update dependency react-router-dom to v7.3.0
  Bump @babel/runtime from 7.26.9 to 7.26.10
  Bump @babel/helpers from 7.26.0 to 7.26.10
  Update dependency @hookform/resolvers to v4
  Update typescript-eslint monorepo to v8.26.0
  Update dependency zod to v3.24.2
  Update dependency eslint-plugin-cypress to v4.2.0
  Update dependency eslint to v9.22.0
  Update dependency eslint-config-prettier to v10.1.1
  Update tanstack-query monorepo to v5.67.2
  Update dependency eslint-plugin-react-hooks to v5.2.0

joshdimanteto committed:
fix linting errors

joshdimanteto committed:
Fix failing test

- Remove once equals from msw function as when a user switches images
  using the thumbnail the level range reset so it need two calls.

- add more assertions for 12 bit and 8 bit images

louise-davies reviewed (approved):
LGTM
